infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -142,7 +156,7 @@ private boolean tableVersionMatchesPlugin() {
             txn.commit();
         }
 
-        return record == null ? false : (Objects.equals(record.getPayload().getVersion(),
+        return record.getPayload() != null && (Objects.equals(record.getPayload().getVersion(),
```

Comment:
dont we need the null check here?

Reply:
getRecord API had a change recently that it won't return null type. Instead, it return an empty record object with null payload. So I changed it to check the payload.